lib/Math/Matrix.pm6

```diff
@@ -51,6 +52,15 @@ method zero(Math::Matrix:U: Positive_Int $rows, Positive_Int $cols = $rows) {
     self.bless( rows => self!zero_array($rows, $cols), row-count => $rows, column-count => $cols );
 }
 
+method submatrix(Math::Matrix:U: Int $row, Int $col --> Math::Matrix:D ){
+    fail "$row is not an existing row index" unless 0 <= $row < $!row-count;
+    fail "$col is not an existing column index" unless 0 <= $col < $!column-count;
+    my @clone = @!rows.clone;
+    @clone.splice($row,1);
+    @clone = map { $^r.splice($col, 1); $^r }, @clone;
+    Math::Matrix.new( @clone );
+}
```

Comment:
my problem is that we in german use completely different names and i did quick lookup. i fully support your proposed function but would call it differently.